### PR TITLE
✨ Placement translator refactor

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR refactors the internal expression of configuration for the last step of the placement translator, to (a) make it less confusing by treating the namespaced and cluster-scoped resources separately and (b) making that configuration follow relaitonal algebra patterns (because that looks like the way to implement it).

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/placement/binding-organizer.go [hunk 1]
```diff
@@ -20,160 +20,222 @@ import (
...
+	logger        klog.Logger
+	discovery     APIMapProvider
+	resourceModes ResourceModes
+	eventHandler  EventHandler
+	// projectionMapProvider TransformingRelayMap[ProjectionKey, *projectionPerClusterImpl, *ProjectionPerCluster]
+	workloadProjector    WorkloadProjector
+	clusterWhatWhereFull MappingReceiver[ClusterWhatWhereFullKey, ProjectionModeVal]
 	sync.Mutex
-}
+	workloadProjectionSections WorkloadProjectionSections // non-nil only during a transaction!
```
Comment (waltforme) [inline pkg/placement/binding-organizer.go:178]: Question: Does this inline comment refer to line 195 (`sbo.workloadProjectionSections = WorkloadProjectionSections{}`)?

Comment (MikeSpreitzer) [reply pkg/placement/binding-organizer.go:178]: yes

Comment (waltforme) [reply pkg/placement/binding-organizer.go:178]: Is `WorkloadProjectionSections{}` itself nil when it has several nil fields? I'm not sure.

Comment (MikeSpreitzer) [reply pkg/placement/binding-organizer.go]: Sorry, typo, the comment should say non-zero rather than non-nil

Comment (MikeSpreitzer) [reply pkg/placement/binding-organizer.go]: Fixed in latest revision.

**Comment by MikeSpreitzer** [comment]
The force-push to 5d253fff is a rebase onto the latest `main` plus more commits working on the real APIMapProvider.

**Comment by MikeSpreitzer** [comment]
@yana1205 
@ezrasilvera

**Pull request merged**